src/control/cmd/dmg/pool.go

```diff
@@ -184,50 +158,27 @@ func (d *PoolDestroyCmd) Execute(args []string) error {
 // PoolQueryCmd is the struct representing the command to destroy a DAOS pool.
 type PoolQueryCmd struct {
 	logCmd
-	connectedCmd
+	ctlClientCmd
 	UUID string `long:"pool" required:"1" description:"UUID of DAOS pool to query"`
 }
 
 // Execute is run when PoolQueryCmd subcommand is activated
 func (c *PoolQueryCmd) Execute(args []string) error {
-	req := client.PoolQueryReq{
+	req := &control.PoolQueryReq{
 		UUID: c.UUID,
 	}
+	req.SetHostList(c.hostlist)
```

Comment:
Yeah, this is kind of awkward, but I think for now I'd prefer not to go down the road of factory methods. The idea is that the hostlist is an optional request parameter, and unfortunately you can't used promoted fields in struct literals, hence the setter method.

I think I'm just going to remove it from these pool commands to avoid future confusion about this point, actually. I was going back and forth about whether or not it should be possible to specify the hostlist from the commandline when making a pool request, but I think I want to set the precedent that it should be handled by the API.

Reply:
All right, that sounds fine to me, letting the API handle it.